### PR TITLE
[PLT-8173] Add username and profile picture to webhook set up pages

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary
This issue exposes a Username and Profile Picture configuration for incoming webhooks to override the defaults:

image

It remains possible to override the Username and Profile Picture when posting the webhook. These settings are tied to the "Enable integrations to override usernames" and "Enable integrations to override profile picture icons" configurations respectively.

#### Ticket Link
https://github.com/mattermost/mattermost-server/issues/7880

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)
- [x] Has server changes: https://github.com/mattermost/mattermost-server/pull/8002
- [x] Has UI changes
- [x] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/mattermost-webapp/blob/master/i18n/en.json)) updates

### CODE REVIEW

File: components/integrations/components/abstract_incoming_webhook.jsx [hunk 1]
```diff
@@ -219,6 +235,64 @@ export default class AbstractIncomingWebhook extends React.Component {
                                 </div>
                             </div>
                         </div>
+                        { global.mm_config.EnablePostUsernameOverride === 'true' &&
```
Comment (lieut-data) [inline components/integrations/components/abstract_incoming_webhook.jsx]: I'd have liked to access the global config via Redux, but opted not to tackle that as part of this PR given that there appears to be no existing reducers for same.

Comment (enahum) [reply components/integrations/components/abstract_incoming_webhook.jsx]: Well we are converting everything to be pureComponents / funtional components and redux connected. You have access in state.entities.general.config so feel free to convert it, actually I would prefer if you do

Comment (jwilander) [reply components/integrations/components/abstract_incoming_webhook.jsx]: It should be under `state.entities.general.config`

Comment (lieut-data) [reply components/integrations/components/abstract_incoming_webhook.jsx]: Oh -- thanks for sharing this! I should have browser the redux state tree itself. I'll update this accordingly.

**Comment by jwilander** [comment]
Once the server PR goes in, I'll ask Linda (QA) to give the UI a test

**Comment by jwilander** [comment]
@lindalumitchell the server PR has been merged. Would you be able to give this a test? The ticket should have enough information for testing but if you have any questions let us know

**Comment by lindalumitchell** [comment]
Thanks @jwilander @lieut-data, looks great! Based on the JIRA ticket info, I verified:

1. Can successfully override username and profile picture as described
2. Payload overrides in-app settings
3. If set in System Console to disallow overrides, username and profile picture post as user who set up the webhook
4. If set in System Console to disallow overrides, webhook setup page hides username and profile picture UI
5. Username field limits input to maximum 22 characters

One question: 
6. If, on creating new webhook, I try to use a character in username that is not allowed (e.g. `$` or `*`), I see `Invalid username` in red and it does not save (as expected). 

However, if I try to edit an existing webhook and add an invalid character to the username, the Update button changes briefly, then no changes are saved (as expected), but there is no feedback printed about why it didn't save. There is help text right there, so maybe that's good enough? Or should we make sure an error prints when editing an existing webhook as well?

**Comment by lieut-data** [comment]
Thanks for such a thorough review, @lindalumitchell! I'll take a look as to why there is no feedback printed about why it didn't save. I agree with you that an error should display in this case.

**Comment by lieut-data** [comment]
Looks like this build is failing on some newly expanded style checks (namely checking `*.js` instead of just `*.jsx`). I'm addressing some naming feedback from https://github.com/mattermost/mattermost-api-reference/pull/315 and will resolve as part of those changes.

**Comment by lieut-data** [comment]
I've rebased off master to get the timezone fixes, addressed feedback from https://github.com/mattermost/mattermost-api-reference/pull/315 (resulting in a new PR to the server at https://github.com/mattermost/mattermost-server/pull/8019), fixed the style checks from the updated rules, and squashed a [bug](https://github.com/mattermost/mattermost-webapp/pull/514/commits/d7485d2eca7a36e09563173b77f358e5c1fa8c11) unrelated to this change that resulted in the missing "Invalid username." text showing up.

I've kept the individual commits for analysis, but of course expect everything to be squashed on merge to `master`.

**Comment by lieut-data** [comment]
@jwilander, can I get another review? Also, given the scope of the changes, I suspect we should recruit @lindalumitchell to run a second test execution.

**Comment by jwilander** [comment]
Thanks @lieut-data, @lindalumitchell mind giving this another quick look?

**Comment by lindalumitchell** [comment]
@lieut-data @jwilander I started to take another look, but now I'm seeing the override settings not saving? Tested on both system admin and non-admin accounts.

1. Integrations > Incoming Webhooks > add new (or edit)
2. Enter values in all fields, including a username and a URL for username and profile picture overrides
3. Click to save
4. Click to Edit that webhook

Observed: The override fields are empty. If you run the webhook, the overrides do not occur
Expected: The values save, and the overrides occur

webhook-overrides-not-saving

**Comment by lieut-data** [comment]
Thanks for your patience with this one, @lindalumitchell :) I tested locally before submitting these changes, but it's likely I misunderstood something about the upgrade process. Let me start by setting up my Mattermost server from a clean state and see if I can reproduce locally.

**Comment by lieut-data** [comment]
Actually, I just realized I could connect to the test server myself, and it /seems/ like the backend isn't running the changes from https://github.com/mattermost/mattermost-server/pull/8019. The webapp is correctly submitting the prefixless parameters (`username` and `icon_url`) but the server is expecting and sending back the old prefixed ones (`post_username` and `post_icon_url`). This would the explain the observations above.

@jwilander, is there a way to confirm what commit is currently deployed to the test instance above, and if necessary, update it accordingly?

**Comment by lieut-data** [comment]
Yes, the server is definitely out of date -- the logs for the above test instance point at commit https://github.com/mattermost/mattermost-server/commit/15cc4497588537f3f9b81d6f97228fae946fa008, the commit just before mine.

I don't have permissions to re-setup a test server as such. @jwilander, @enahum, @crspeller -- help?

**Comment by crspeller** [comment]
@lieut-data @lindalumitchell Should be updated now.

**Comment by lieut-data** [comment]
Thanks, @crspeller! @lindalumitchell, I've verified that the override settings are now saving on the test server in tandem with the server changes being correctly deployed. Sorry for the churn :)

**Comment by lindalumitchell** [comment]
Thanks! And sorry I just inadvertently destroyed the new spinmint; setting up a new-new one now and will test. :)

**Comment by lindalumitchell** [comment]
@lieut-data @jwilander I just re-tested everything, and it all looks great! No issues found. Thanks much!

**Pull request merged**